BookTracking.Web/src/pages/Home/BookTable.tsx

```diff
@@ -0,0 +1,119 @@
+import { Fragment, ReactNode, useState } from 'react';
+import { Anchor, Badge, Button, Collapse, Flex, Image, Table } from '@mantine/core';
+import { useDisclosure } from '@mantine/hooks';
+import { Book } from '../../models/Book';
+import { ExpandedBookRow } from './ExpandedBookRow';
+
+import './BookTable.css';
+
+const rowData: Book[] = [
+  {
+    id: '1',
+    authorId: '1',
+    link: new URL('https://www.google.com'),
+    coverImage: './src/assets/testImage.webp',
+    title: 'Book Title',
+    author: 'Author',
+    tags: ['Fantasy', 'Magic'],
+    rating: 7,
+  },
+  {
+    id: '2',
+    authorId: '2',
+    link: new URL('https://www.google.com'),
+    coverImage: './src/assets/testImage.webp',
+    title: 'Book Title 2',
+    author: 'Author',
+    tags: ['Fantasy', 'Magic'],
+    rating: 8,
+  },
+  {
+    id: '3',
+    authorId: '3',
+    link: new URL('https://www.google.com'),
+    coverImage: './src/assets/testImage.webp',
+    title: 'Book Title 3',
+    author: 'Author',
+    tags: ['Fantasy', 'Magic', 'Sci-Fi', 'History', 'Progression', 'Cultivation', 'Fantasy'],
+    rating: 9,
+  },
+  {
+    id: '4',
+    authorId: '4',
+    link: new URL('https://www.google.com'),
+    coverImage: './src/assets/testImage.webp',
+    title: 'Book Title 4',
+    author: 'Author',
+    tags: ['Fantasy', 'Magic'],
+    rating: 10,
+    lastReadBookOrChapter: 'Ch. 23',
+  },
+];
+
+export const BookTable = () => {
+  const [openedRows, setOpenedRows] = useState<number[]>([]);
+
+  const toggleOpened = (index: number) => {
```

Comment:
Two things:
1) `toggledOpened` is passed to a child component, and should use the `useCallback` hook. Currently every render this function is re-created, and it will cause the `<Button>` that uses it to re-render because it detects this as a prop change. In general any in-component function you write and pass to children should use [useCallback](https://react.dev/reference/react/useCallback) to make it a stable function.
2) `setOpenedRows` should probably track `book.id` instead of index. By tacking index the opened rows can change based on render order instead of book opened. ex: if the first book expanded (titled "Book Title"), and you reverse the sort of the list, the new book at index 0 would be expanded (titled "Book Title 4").